accepted/0004-undo-redo.md

```diff
@@ -0,0 +1,73 @@
+- Start Date: 2020-01-03
+- RFC PR: #5
+- Mantis Issue: N/A
+
+# Summary
+Create a undo system that is capable of recording the state of OBS actions to
+easily revert the state forwards or backwards in time.
+
+# Motivation
+When working with large and intricate processes - whether it be scenes, sources,
+etc. - accidents are bound to happen. Currently, there is no way to undo
+committed actions. As a result, work can easily be lost at the expense of hours
+of effort.
+
+Implementing a proper undo/redo system would be a drastic increase in
+quality of life for the majority of users.
+
+# Design
+## UX
+The UX would model that of most other applications. Under edit, there would be 2
+options: 'Undo' and 'Redo'. When an action is comitted, for instance deleting a
+scene, the undo button becomes activated, allowing the user to select it.
+
+When activated, it undos the action and then activates the Redo button, which
+allows the user to essentially 'undo' their previous undo. If the user makes
+multiple undos, then the users would also have multiple redos available.
+
+When a user undos and then commits another action, the redos are no longer
+possible as the tree has diverged into a new path.
+
+Also, like many other applications, the undo/redo keys should be binded to Ctrl-Z
+and Ctrl-Y
+
+## Functionality
+### Core Functionality
+The undo/redo system work by saving states. However, saving the entire state of
+OBS would use up more memory than necessary when a small action is committed. In
```

Comment:
Keeping the memory footprint low is ideal, and I completely agree with only storing the exact values that changed. Are we thinking of a maximum size of the undo stack, either by number of actions or used memory?

Reply:
I'd think number of actions should probably be fine, even 100 might be enough.